indra/llmessage/llavatarnamecache.cpp

```diff
@@ -179,19 +180,29 @@ void LLAvatarNameCache::requestAvatarNameCache_(std::string url, std::vector<LLU
 
         if (LLAvatarNameCache::instanceExists())
         {
-            if (!success)
-            {   // on any sort of failure add dummy records for any agent IDs 
-                // in this request that we do not have cached already
-                std::vector<LLUUID>::const_iterator it = agentIds.begin();
-                for (; it != agentIds.end(); ++it)
-                {
-                    const LLUUID& agent_id = *it;
-                    LLAvatarNameCache::getInstance()->handleAgentError(agent_id);
-                }
-                return;
-            }
+            // dispatch handler execution back to mainloop
+            auto workqueue = LL::WorkQueue::getInstance("mainloop");
 
-            LLAvatarNameCache::getInstance()->handleAvNameCacheSuccess(results, httpResults);
+            if (workqueue)
+            {
+                workqueue->post([=]()
+                    {
+
+                        if (!success)
+                        {   // on any sort of failure add dummy records for any agent IDs 
+                            // in this request that we do not have cached already
+                            std::vector<LLUUID>::const_iterator it = agentIds.begin();
+                            for (; it != agentIds.end(); ++it)
```

Comment:
`for (const auto& agent_id : agent_ids)`

Reply:
I just indented it, but sure...